### PR TITLE
8271946: Cleanup leftovers in Space and subclasses

### PR DESCRIPTION
**Pull request by rkennke**
There are a number of methods in (Compactible)Space that seem trivially implemented. E.g. obj_size(), adjust_obj_size(), scanned_block_is_obj() etc. I believe they are leftovers from CMS (not sure) but looks like they can be replaced by obj->size() or removed altogether. Also, the the related templated methods scan_and_forward(), scan_and_adjust_pointers() and scan_and_compact() can be directly and mechanically inlined into their callers. It makes the code more readable.

Testing:
 - [x] hotspot_gc
 - [x] tier1
 - [x] tier2

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8271946](https://bugs.openjdk.java.net/browse/JDK-8271946): Cleanup leftovers in Space and subclasses

### Reviewers
 * [Stefan Karlsson](https://openjdk.java.net/census#stefank) (@stefank - **Reviewer**)
 * [Thomas Schatzl](https://openjdk.java.net/census#tschatzl) (@tschatzl - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/5031/head:pull/5031` \
`$ git checkout pull/5031`

Update a local copy of the PR: \
`$ git checkout pull/5031` \
`$ git pull https://git.openjdk.java.net/jdk pull/5031/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 5031`

View PR using the GUI difftool: \
`$ git pr show -t 5031`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/5031.diff">https://git.openjdk.java.net/jdk/pull/5031.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/gc/shared/space.hpp [hunk 1]
```diff
@@ -312,11 +312,6 @@ class CompactPoint : public StackObj {
 // The compaction operations are implemented by the
 // scan_and_{adjust_pointers,compact,forward} function templates.
 // The following are, non-virtual, auxiliary functions used by these function templates:
-// - scan_limit()
-// - scanned_block_is_obj()
-// - scanned_block_size()
-// - adjust_obj_size()
-// - obj_size()
```
Comment (stefank) [inline src/hotspot/share/gc/shared/space.hpp:319]: The entire surrounding comment needs to be updated now that these functions have been removed.

**Review by stefank — APPROVED** [review]

**Review by stefank — APPROVED** [review]

**Comment by rkennke** [comment]
Ping? Anybody else who wants to approve this? Thanks!

**Review by tschatzl — APPROVED** [review]
Lgtm.

**Pull request closed without merging**